src/package.cc

```diff
@@ -52,9 +68,34 @@ std::pair<double, int> Package::GetFillMass(double qty) {
       fill_mass = fill_max_;
     }
   }
-  fill_mass = std::min(qty, fill_mass);
-  num_at_fill_mass = static_cast<int>(std::floor(qty / fill_mass));
-  return std::pair<double, int>(fill_mass, num_at_fill_mass);
+
+  if (strategy_ == "first" || strategy_ == "equal") {
+    fill_mass = std::min(qty, fill_mass);
+    num_at_fill_mass = static_cast<int>(std::floor(qty / fill_mass));
+    ExceedsSplitLimits(num_at_fill_mass);
+    packages.assign(num_at_fill_mass, fill_mass);
+
+    qty -= num_at_fill_mass * fill_mass;
+  }
+
+  if (strategy_ == "uniform" || strategy_ == "normal") {
+    // only use random if a full package amount is available. if less than one
+    // full amount is available, below will fill a partial package (no random).
+    while ((qty > 0) && (qty >= fill_max_)) {
```

Comment:
aren't these two conditions redundant?  Or maybe cleaner to do:
```suggestion
    while ( qty > std::max(0,fill_max_) ) {
```